src/lib/pubkey/xmss/xmss_tools.h

```diff
@@ -0,0 +1,111 @@
+/**
+ * XMSS Address
+ * (C) 2016 Matthias Gierlings
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ **/
+
+#ifndef BOTAN_XMSS_TOOLS_H__
+#define BOTAN_XMSS_TOOLS_H__
+
+#include <stdint.h>
+#include <iterator>
+#include <type_traits>
+#include <botan/types.h>
+#include <botan/secmem.h>
+
+namespace Botan {
+
+/**
+ * Helper tools for low level byte operations required
+ * for the XMSS implementation.
+ **/
+ class XMSS_Tools
```

Comment:
This seems like a lot of extra syntax just to save `m_is_little_endian` - but there was no runtime endian check before in utils. I've added `CPUID::is_little_endian()` to master in 523b2a4ca, cpuid.h header is public. It seems with that call `XMSS_Tools` class/singleton could be removed and the `concat`s do not need to be member functions anymore, also simplifying syntax for callers.